src/Hyperion/Main.hs

```diff
@@ -130,22 +134,26 @@ doRun bks = do
       throwIO $ DuplicateNames [ n | n:_:_ <- group (sort nms) ]
     foldMap (runBenchmark (sample 100 (fixed 5))) bks
 
-doAnalyze :: Config -> [Benchmark] -> IO ()
-doAnalyze Config{..} bks = do
+doAnalyze :: Config -> Text -> [Benchmark] -> IO ()
+doAnalyze Config{..} benchName bks = do
+    executableName <- getProgName -- Name of the execuatble that launched the benches.
     h <- case configOutputPath of
       Nothing -> return IO.stdout
-      Just path -> IO.openFile path IO.WriteMode
+      Just path -> do
+        let filename = executableName <.> (unpack benchName) <.> "json"
+        createDirectoryIfMissing True path -- Creates the directory if needed.
+        IO.openFile (path </> filename) IO.WriteMode
     results <- doRun bks
     let strip
           | configRaw = id
           | otherwise = reportMeasurements .~ Nothing
-        report = results & imapped %@~ analyze & mapped %~ strip
+        report = results & imapped %@~ (analyze (pack executableName)) & mapped %~ strip
```

Comment:
This looks incorrect to me: you're passing the `executableName` as the `packageName`.